x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+import uuid from 'uuid/v4';
+import { PromiseReturnType } from '../../../../observability/typings/common';
+import { Setup } from '../helpers/setup_request';
+import { JobResponse } from '../../../../ml/common/types/modules';
+import {
+  SERVICE_ENVIRONMENT,
+  TRANSACTION_DURATION,
+  PROCESSOR_EVENT,
+} from '../../../common/elasticsearch_fieldnames';
+
+const ML_MODULE_ID_APM_TRANSACTION = 'apm_transaction';
+export const ML_GROUP_NAME_APM = 'apm';
+
+export type CreateAnomalyDetectionJobsAPIResponse = PromiseReturnType<
+  typeof createAnomalyDetectionJobs
+>;
+export async function createAnomalyDetectionJobs(
+  setup: Setup,
+  environments: string[],
+  logger: Logger
+) {
+  const { ml, config } = setup;
+  if (!ml) {
+    return [];
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (
+    !(
+      mlCapabilities.mlFeatureEnabledInSpace &&
+      mlCapabilities.isPlatinumOrTrialLicense
+    )
+  ) {
+    logger.warn(
+      'Anomaly detection integration is not available for this user.'
+    );
+    return [];
+  }
+  logger.info(
+    `Creating ML anomaly detection jobs for environments: [${environments}].`
+  );
+
+  const indexPatternName = config['apm_oss.transactionIndices']; // TODO [ML] - Do we want to use the config index name?
+  const dataRecognizerConfigResponses = await Promise.all(
+    environments.map((environment) =>
+      configureAnomalyDetectionJob({ ml, environment, indexPatternName })
```

Comment:
Is there a difference between "configure job" and "create job"? If not, let's stick to `createAnomalyDetectionJob`